src/Compute/Metadata/Readers/StreamReader.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/**
+ * Copyright 2015 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+namespace Google\Cloud\Compute\Metadata\Readers;
+
+/**
+ * A class only reading the metadata URL with an appropriate header.
+ *
+ * This class makes it easy to test the MetadataStream class.
+ */
+class StreamReader
+{
+    /**
+     * The header whose presence indicates GCE presence.
+     */
+    const FLAVOR_HEADER = 'Metadata-Flavor';
+
+    public function read($url)
+    {
+        $options = array(
+            'http' => array(
+                'method' => 'GET',
+                'header' => self::FLAVOR_HEADER.": Google\r\n",
```

Comment:
Could we assign the header as a whole as a const instead of doing concatenation here?